src/main/java/mimon/Mimon.java

```diff
@@ -0,0 +1,204 @@
+package mimon;
+import tasks.Task;
+import tasks.Todo;
+import tasks.Deadline;
+import tasks.Event;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Mimon {
+    private static final String FILE_PATH = "./data/mimon.txt";
+    public static final String DASHED_LINE = "____________________________________________________________";
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static Scanner scanner = new Scanner(System.in);
+
+    public static void main(String[] args) {
+        loadTasksFromFile();
+        printWelcomeMessage();
+
+        while (true) {
+            String userInput = scanner.nextLine().trim();
+
+            if (userInput.equalsIgnoreCase("bye")) {
+                printExitMessage();
+                break;
+            }
+
+            try {
+                processUserCommand(userInput);
+                saveTasksToFile();
+            } catch (MimonException e) {
+                System.out.println(DASHED_LINE);
+                System.out.println("Hold on! " + e.getMessage());
+                System.out.println(DASHED_LINE);
+            }
+        }
+    }
+
+    private static void processUserCommand(String userInput) throws MimonException {
+        if (userInput.equalsIgnoreCase("list")) {
+            listTasks();
+        } else if (userInput.startsWith("mark ")) {
+            markTask(userInput, true);
+        } else if (userInput.startsWith("unmark ")) {
+            markTask(userInput, false);
+        } else if (userInput.startsWith("delete ")) {
+            deleteTask(userInput);
+        } else if (userInput.startsWith("todo")) {
+            if (userInput.trim().equals("todo")) {
+                throw new MimonException("The description of a todo must have some information.");
+            }
+            addTodoTask(userInput.substring(5).trim());
+        } else if (userInput.startsWith("deadline")) {
+            if (userInput.trim().equals("deadline")) {
+                throw new MimonException("The description of a deadline should be meaningful.");
+            }
+            addDeadlineTask(userInput.substring(9).trim());
+        } else if (userInput.startsWith("event")) {
+            if (userInput.trim().equals("event")) {
+                throw new MimonException("The description of an event should include the date!.");
+            }
```

Comment:
This is ~3 levels of indentation. I suggest you refactor the innermost levels into functions of their own to improve the readability of this method.